CDP4Reporting/ViewModels/ReportDesignerViewModel.cs

```diff
@@ -644,11 +657,40 @@ private async Task ExecuteRebuildDatasourceCommand()
                 this.IsBusy = false;
             }
         }
+        
+        /// <summary>
+        /// Executes the <see cref="RebuildDatasourceAndRefreshPreviewCommand"/>
+        /// </summary>
+        /// <returns>An awaitable <see cref="Task"/></returns>
+        private async Task ExecuteRebuildDatasourceAndRefreshPreviewCommand()
+        {
+            this.IsBusy = true;
+
+            try
+            {
+                var source = this.Document.Text;
+                await this.compilationConcurrentActionRunner.RunAction(() => this.CompileAssembly(source));
+
+                if (this.CompileResult?.Errors.HasErrors ?? false)
+                {
+                    this.messageBoxService.Show(this.CompileResult?.Errors.ToString(), "Compilation error", MessageBoxButton.OK, MessageBoxImage.Stop);
+                }
+
+                this.RebuildDataSource(true);
+                var presenter = this.currentReportDesignerDocument?.Preview as DocumentPreviewControl;
+                presenter?.ParametersModel.SubmitParameters();  
+            }
```

Comment:
perhaps here we  can log any error that has occurred?

Reply:
These errors are "temporary", because they're build/compile errors. 
These errors are also shown in the the specific error panel  in the report designer.
Therefore I would suggest not to log them like a "normal" error. 
The extra messagebox is there, because the focus of the previewer is different (on the end result and not on the C# code) and is only meant to notify the user that something went wrong and the report probably shows the wrong data.

Reply:
not even a Log.Trace?

Reply:
No, I would suggest not to do that.